### PR TITLE
Use Vercel Edge Functions

### PR DESCRIPTION
**Pull request by zeke**
This PR updates the API routes to use Vercel Edge Functions

From https://vercel.com/docs/concepts/functions/edge-functions

> Our Edge Runtime is more performant and cost-effective than Serverless Functions on average. Edge Functions are deployed globally on our Edge Network, and can automatically execute in the region nearest to the user who triggers them. They also have no cold boots, which means they don't need extra time to start up before executing your code.

## References

- https://github.com/vercel/examples/pull/622 - 🎩  hat tip to @steven-tey 
- https://nextjs.org/docs/api-reference/next/server#nextrequest
- https://nextjs.org/docs/api-reference/next/server#nextresponse
- https://developer.mozilla.org/en-US/docs/Web/API/Request
- https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream
- https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams

### CODE REVIEW

File: pages/api/predictions/index.js [hunk 1]
```diff
@@ -6,36 +7,43 @@ const replicate = new Replicate({
   userAgent: `${packageData.name}/${packageData.version}`,
 });
 
+async function getObjectFromRequestBodyStream(body) {
```
Comment (zeke) [inline pages/api/predictions/index.js:10]: This works but it doesn't feel right. What's a better way to do this?

Update `replicate` to accept a ReadableStream as `input`?

Comment (mattt) [reply pages/api/predictions/index.js:10]: I think `const input = await req.body.json()` could work. Could you try that?

Comment (zeke) [reply pages/api/predictions/index.js:10]:
```
error - req.body.json is not a function
```

Comment (mattt) [reply pages/api/predictions/index.js:10]: Ah well, thanks for trying that out. As cumbersome as this is, all of the other solutions I've found are even worse, somehow.

As far as putting this in the client, I'd imagine that most consumers wouldn't be passing full request bodies directly, so accepting a `ReadableStream` probably wouldn't be helpful.

Comment (zeke) [reply pages/api/predictions/index.js:10]: I opened an issue on the Next.js repo for guidance about best practice here: https://github.com/vercel/next.js/issues/48403

In the meantime I'll go ahead and ship this and we can come back if we discover a better approach.

**Pull request merged**